src/raftstore/store/peer_storage.rs

```diff
@@ -994,54 +1032,79 @@ impl PeerStorage {
 }
 
 /// Delete all meta belong to the region. Results are stored in `wb`.
-pub fn clear_meta(engine: &DB,
-                  wb: &WriteBatch,
+pub fn clear_meta(wb: &WriteBatch,
+                  raft_wb: &WriteBatch,
+                  raft_engine: &DB,
                   region_id: u64,
-                  raft_state: &RaftLocalState)
+                  raft_state: &RaftLocalState,
+                  is_recovery: bool)
```

Comment:
so the is_recovery = true is only for tombstone clean up? If yes, recovery here is not proper.